typed constant

The following program `sample.go` triggers a panic:

```go
package main
  
import (
    "fmt"
    "time"
)

const df = time.Minute * 30

func main() {
    fmt.Printf("df: %v %T\n", df, df)
}
```

Expected result:
```console
$ go run ./sample.go
df: 30m0s time.Duration
```

Got:
```console
$ yaegi ./sample.go
./sample.go:8:12: panic
panic: reflect.Set: value of type int is not assignable to type time.Duration [recovered]
        panic: reflect.Set: value of type int is not assignable to type time.Duration

goroutine 1 [running]:
github.com/containous/yaegi/interp.runCfg.func1(0xc0000d1680, 0xc0000daf00)
        /home/marc/go/src/github.com/containous/yaegi/interp/run.go:112 +0x218
panic(0xc9ba60, 0xc00013a660)
        /usr/lib/go/src/runtime/panic.go:679 +0x1b2
reflect.Value.assignTo(0xc9a960, 0xc000034b80, 0x182, 0xdda212, 0xb, 0xd69f80, 0x0, 0xc000195c10, 0x444cd3, 0xd69f80)
        /usr/lib/go/src/reflect/value.go:2403 +0x432
reflect.Value.Set(0xd69f80, 0xc000034b88, 0x186, 0xc9a960, 0xc000034b80, 0x182)
        /usr/lib/go/src/reflect/value.go:1537 +0xbd
github.com/containous/yaegi/interp.assign.func3(0xc0000d1680, 0xc00000f300)
        /home/marc/go/src/github.com/containous/yaegi/interp/run.go:301 +0xb6
github.com/containous/yaegi/interp.runCfg(0xc0000daf00, 0xc0000d1680)
        /home/marc/go/src/github.com/containous/yaegi/interp/run.go:118 +0x7a
github.com/containous/yaegi/interp.(*Interpreter).run(0xc000244580, 0xc0000da400, 0x0)
        /home/marc/go/src/github.com/containous/yaegi/interp/run.go:96 +0x239
github.com/containous/yaegi/interp.(*Interpreter).Eval(0xc000244580, 0xc0000ba280, 0x79, 0x279, 0xc0000ba280, 0x79, 0x0, 0x1)
        /home/marc/go/src/github.com/containous/yaegi/interp/interp.go:364 +0x3ef
main.main()
        /home/marc/go/src/github.com/containous/yaegi/cmd/yaegi/yaegi.go:82 +0x42e
```

--- a/_test/time11.go
+++ b/_test/time11.go
@@ -1,0 +1,15 @@
+package main
+
+import (
+	"fmt"
+	"time"
+)
+
+const df = time.Minute * 30
+
+func main() {
+	fmt.Printf("df: %v %T\n", df, df)
+}
+
+// Output:
+// df: 30m0s time.Duration

--- a/interp/cfg.go
+++ b/interp/cfg.go
@@ -602,8 +602,6 @@ func (interp *Interpreter) cfg(root *node) ([]*node, error) {
 				constOp[n.action](n)
 			}
 			switch {
-			//case n.typ != nil && n.typ.cat == BoolT && isAncBranch(n):
-			//	n.findex = -1
 			case n.rval.IsValid():
 				n.gen = nop
 				n.findex = -1
@@ -1851,14 +1849,10 @@ func arrayTypeLen(n *node) int {
 
 // isValueUntyped returns true if value is untyped
 func isValueUntyped(v reflect.Value) bool {
-	// Consider only untyped constant values.
+	// Consider only constant values.
 	if v.CanSet() {
 		return false
 	}
-	// Consider only values of default numerical types.
-	switch v.Type().Kind() {
-	case reflect.Int, reflect.Int32, reflect.Int64, reflect.Uint32, reflect.Uint64, reflect.Float64, reflect.Complex128:
-		return true
-	}
-	return false
+	t := v.Type()
+	return t.String() == t.Kind().String()
 }

